neurodamus/ngv.py

```diff
@@ -11,65 +10,114 @@
-from .core.configuration import GlobalConfig, LogLevel
+from .core.configuration import ConfigurationError, GlobalConfig, LogLevel
 from .io.sonata_config import ConnectionTypes
 from .io.synapse_reader import SonataReader, SynapseParameters
 from .metype import BaseCell
 from .morphio_wrapper import MorphIOWrapper
-from .utils.logging import log_verbose
-from .utils.pyutils import append_recarray, bin_search
+from .utils.pyutils import append_recarray
 
 
 class Astrocyte(BaseCell):
-    __slots__ = ("_nseg_warning", "section_names", "sections_glut", "soma_glut")
+    __slots__ = ("_gluts", "glut_soma", "section_names")
 
     def __init__(self, gid, meinfos, circuit_conf):
-        """Initialize an Astrocyte cell"""
+        """Initialize an Astrocyte cell:
+
+        - create the cell from Cell.hoc
+        - add the morphology
+        - resize sections if necessary
+        - add cadifus, GlutReceive (they will be connected later)
+        - record section_names for creating connections later
+        """
         super().__init__(gid, meinfos, None)
 
-        # Compose the path to the morphology file
-        morph_file = (
+        # Create the cell
+        self._cellref = Nd.Cell(gid)  # cell instantiated with Cell.hoc
+        # load and apply morphology
+        morph = MorphIOWrapper(
             Path(circuit_conf.MorphologyPath)
             / f"{meinfos.morph_name}.{circuit_conf.MorphologyType}"
         )
-
-        # Create the cell and load morphology
-        self._cellref = Nd.Cell(gid)
-        morph = MorphIOWrapper(morph_file)
         self._cellref.AddHocMorph(morph.morph_as_hoc())
-
-        self._nseg_warning = 0
-
         # Recalculate number of segments and sections
         self._cellref.geom_nseg_fixed()
         self._cellref.geom_nsec()
 
         logging.debug("Instantiating NGV cell gid=%d", gid)
 
-        # Insert mechanisms and glutamate receptors in each section
-        self.sections_glut = []
-        for sec in self._cellref.all:
-            if sec.nseg > 1:
-                self._nseg_warning = 1
-                sec.nseg = 1
-            sec.insert("cadifus")
-            glut = Nd.GlutReceive(sec(0.5), sec=sec)
-            Nd.setpointer(glut._ref_glut, "glu2", sec(0.5).cadifus)
-            self.sections_glut.append(glut)
-
-        # Configure endoplasmic reticulum and section parameters
-        self._cellref.execute_commands(self._er_as_hoc(morph))
-        self._cellref.execute_commands(self._secparams_as_hoc(morph))
-
-        # Soma-specific glutamate receptor (must be last)
-        # used only for accounting for metabolsim. It should not be
-        # connected to other point processes or mechanisms
+        # assigned later
+        self._gluts = {}
+        is_resized_secs = False
+        for sec in self.all:
+            is_resized = self._init_basic_section(sec)
+            is_resized_secs = is_resized_secs or is_resized
+
+        if is_resized_secs:
+            logging.warning(
```

Comment:
As discussed, this *might* show many warnings. We could reduce it more if it is worth the effort? I won't press too much at this time, though.

Reply:
I think a natural development is that we accumulate the warnings and emit at the end of init instead of an `ad hoc` function `post_stdinit`. I can investigate that